### ISSUE TITLE
sbt 1.4.0/1.4.1 ~reStart with sbt-revolver requires "enter" key between "build triggered" and "compiling" if sbt.log.noformat=true

### ISSUE THREAD

**Issue by ritschwumm** [sbt/sbt#6013]
## steps

create a minimal project like this:

```
// build.sbt
organization := "de.test"
name := "repro"
version := "0.0.1"
scalaVersion := "2.13.3"
```
```
// project/plugins.sbt
addSbtPlugin("io.spray" % "sbt-revolver" % "0.9.1")
```
```
# project/build.properties
sbt.version=1.4.1
```
```
// src/main/scala/Repro.scala
object Repro {
	def main(args:Array[String]):Unit	= {
		println("test")
	}
}
```

now start revolver with `~reStart` (in my case, in the jEdit terminal where `TERM=dumb`, but a regular bash seems to behave in the same way). this compiles the code and prints "test" to the terminal.

now change the string printed in Repro.scala in your editor.

## problem

when scala code is changed a line like this is printed
```[info] Build triggered by .../repro/src/main/scala/Repro.scala. Running 'reStart'```

then, the build hangs until i press the <RETURN> key.

when i do, i get the expected log output
```[info] compiling 1 Scala source to .../repro/target/scala-2.13/classes ..```
and the new main class is run and prints to the terminal.

## expectation

i'd expect compilation to happen immediately after the `Build triggered by` message, without waiting for a key press, and the main class of the project to be run,  like it did in earlier sbt versions

**Comment by francisdb** [comment]
Your example works correctly here, no input needed  (mac/iterm2/zsh)

`zsh 5.7.1 (x86_64-apple-darwin19.0)`

```
TERM=xterm-256color
TERM_PROGRAM=iTerm.app
TERM_PROGRAM_VERSION=3.3.12
LC_CTYPE=UTF-8
LC_TERMINAL=iTerm2
LC_TERMINAL_VERSION=3.3.12
```

```
...
[info] Build triggered by /.../test/src/main/scala/Repro.scala. Running 'reStart'.
[info] compiling 1 Scala source to /.../test/target/scala-2.13/classes ...
[info] Application test not yet started
[info] Starting application test in the background ...
test Starting Repro.main()
[success] Total time: 0 s, completed 23 Oct 2020, 15:08:43
[info] 2. Monitoring source files for test/reStart...
[info]    Press <enter> to interrupt or '?' for more options.
test test2
test ... finished with exit code 0
[info] Build triggered by /.../test/src/main/scala/Repro.scala. Running 'reStart'.
[info] compiling 1 Scala source to /.../test/target/scala-2.13/classes ...
[info] Application test not yet started
[info] Starting application test in the background ...
test Starting Repro.main()
[success] Total time: 0 s, completed 23 Oct 2020, 15:08:49
[info] 3. Monitoring source files for test/reStart...
[info]    Press <enter> to interrupt or '?' for more options.
test test1
test ... finished with exit code 0
```

**Comment by ritschwumm** [comment]
thanks @francisdb for trying out, it seems i missed a vital detail:

the problem only occurs when starting sbt like this
```~/Downloads/sbt/bin/sbt -Dsbt.log.noformat=true```
but does _not_ occur when starting it without the noformat property:
```~/Downloads/sbt/bin/sbt```

on a side note: while sbt is waiting for me to press enter, even `ctrl-c` does not kill the process any more

i'm on ubuntu focal, btw.

**Comment by eatkins** [comment]
Thanks for the report! This was actually already fixed before the issue was opened but after 1.4.1 by a892a8368144c4d1455baf5491da42d6152a0658.
